intentgen/src/main/java/com/chatbot/IntentGenerator.java

```diff
@@ -0,0 +1,190 @@
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.chatbot.protobuf.UseCaseOuterClass.UseCase;
+import com.google.cloud.dialogflow.v2.Intent;
+import com.google.cloud.dialogflow.v2.Context;
+import com.google.cloud.dialogflow.v2.Intent.TrainingPhrase;
+import com.google.cloud.dialogflow.v2.Intent.TrainingPhrase.Part;
+import com.google.cloud.dialogflow.v2.IntentsClient;
+import com.google.cloud.dialogflow.v2.ProjectAgentName;
+import com.google.protobuf.TextFormat;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+public class IntentGenerator {
+  static final List<String> YES_TRAINING_PHRASES = new ArrayList<String>(Arrays.asList("yes",
+      "okay I will", "why not", "yes that's alright", "yes I do", "exactly", "of course",
+      "yep that's ok", "okay", "ok", "sure"));
+  static final List<String> NO_TRAINING_PHRASES = new ArrayList<String>(Arrays.asList("No",
+      "thanks but no", "no way", "no no don't", "na", "no it isn't", "don't", "nah I'm good",
+      "no I cannot", "I can't"));
+
+  public static void main(final String[] args) throws ParseException {
+    final String useCaseFile = getUseCaseFile(args);
+    final String projectID = System.getenv("projectID");
+    final ProjectAgentName parent = ProjectAgentName.of(projectID);
+    try(IntentsClient intentsClient = IntentsClient.create()) {
+      // this map is needed to check if an intent display name already exists and if so, instead of 
+      // creating an intent (which would return an error) we update the intent using the name
+      final Map<String, String> intentDisplayNameToName = getIntentDisplayNametoNameMapping(intentsClient,
+          parent);
+      final List<Intent> intentProtobufList = getIntentsInUseCase(useCaseFile).stream()
+          .map(x->intentToIntentProtobuf(x, intentDisplayNameToName, projectID))
+          .collect(Collectors.toList());
+      createOrUpdateIntents(intentProtobufList, intentsClient, intentDisplayNameToName, parent);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  public static TrainingPhrase encodedStringToTrainingPhrase(String trainingPhrase) {
+    TrainingPhrase.Builder trainingPhraseProtobufBuilder = TrainingPhrase.newBuilder();
+    final String[] trainingPhraseParts = trainingPhrase.split("\\|");
+    List<Part> partProtobufList = new ArrayList<Part>();
+    for(String part: trainingPhraseParts) {
+      Part partProtobuf;
+      // semicolon implies that this part is an entity
+      if(part.indexOf(";") != -1) {
+        final String[] entityParts = part.split(";");
+        partProtobuf = Part.newBuilder()
+        .setEntityType(entityParts[0])
+        .setAlias(entityParts[1])
+        .setText(entityParts[2])
```

Comment:
Use constants with names instead of  0, 1, 2.

Reply:
updated